stock_inventory_lockdown/models/stock_inventory.py

```diff
@@ -0,0 +1,29 @@
+# -*- coding: utf-8 -*-
+# © 2013-2016 Numérigraphe SARL
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from openerp import models, api
+
+
+class StockInventory(models.Model):
+    _inherit = 'stock.inventory'
+
+    @api.model
+    def _get_locations_open_inventories(self):
```

Comment:
@mreficent IMO if would be more efficient to allow to pass a location_ids parameter to this method and t use this paramaterd  (if provided) into the domain for the search on inventories...
ex:
```python
 @api.model
 def _get_locations_open_inventories(self, locations_ids = None):
    inventory_domain = [('state', '=', 'confirm')]
    if locations_ids:
        inventory_domain.append(('location_id', 'child_of', location_ids))
    inventories = self.search(inventory_domain)
    location_ids = inventories.mapped('location_id')
    location_domain = [
        ('location_id', 'child_of', location_ids.ids),
        ('usage', 'in', ['internal', 'transit'])]
     if locations_ids:
        location_domain.append(('location_id', 'child_of', location_ids))
    return self.env['stock.location'].search(location_domain)
```
In this way we will filter the result to the locations for which we want to check if an inventory exist.